state_synchronizer/src/peer_manager.rs

```diff
@@ -68,12 +82,33 @@ impl PeerManager {
         self.get_active_upstream_peer_ids().is_empty()
     }
 
+    pub fn update_score(&mut self, peer_id: &PeerId, update_type: PeerScoreUpdateType) {
+        if let Some(peer_info) = self.peers.get_mut(peer_id) {
+            match update_type {
+                PeerScoreUpdateType::Success => {
+                    if peer_info.score < 100 {
+                        peer_info.score += 1;
+                    }
+                }
+                PeerScoreUpdateType::InvalidChunk => {
+                    peer_info.score -= 1;
```

Comment:
I think we should penalize malicious behavior more then award good chunks. So let's say. in case of InvalidChunk, you decrement score by 20, but for `Success` increase it by 1 only
Also make sure that score is >= 1 so we never ran out of sync peers

Reply:
using a additive increase/multiplicative decrease is a fairly standard practice in these scenarios. We might also want to abstract the scoring mechanism into a separate struct (https://en.wikipedia.org/wiki/Separation_of_mechanism_and_policy)